cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -0,0 +1,111 @@
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/repository")
+public class SourceControlManagementHttpHandler extends AbstractAppFabricHttpHandler {
+  private final SourceControlManagementService sourceControlService;
+  private static final Gson GSON = new Gson();
+
+  @Inject
+  SourceControlManagementHttpHandler(SourceControlManagementService sourceControlService) {
+    this.sourceControlService = sourceControlService;
+  }
+
+  @PUT
+  @Path("/")
+  @AuditPolicy(AuditDetail.REQUEST_BODY)
+  public void setRepository(FullHttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId) throws Exception {
+    RepositoryConfigRequest repoRequest = getRepository(request);
+    if (!repoRequest.shouldValidate()) {
+      if (!repoRequest.isValid()) {
+        throw new BadRequestException(String.format("Invalid repository configuration: %s.", repoRequest));
+      }
+      sourceControlService.setRepository(validateNamespaceId(namespaceId), repoRequest);
+      responder.sendString(HttpResponseStatus.OK, String.format("Updated repository configuration for namespace '%s'.",
+                                                                namespaceId));
+    }
+
+    // TODO: CDAP-20252 add the validate logic once the SourceControlManager module is ready
+  }
+
+  @GET
+  @Path("/")
+  public void getRepository(FullHttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId) throws Exception {
+    NamespaceId namespace = validateNamespaceId(namespaceId);
+    RepositoryConfig repository = sourceControlService.getRepository(namespace);
+
+    if (repository == null) {
+      throw new RepositoryNotFoundException(namespace);
+    }
+
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(repository));
+  }
+
+  @DELETE
+  @Path("/")
+  public void deleteRepository(FullHttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId) throws Exception {
+    sourceControlService.deleteRepository(validateNamespaceId(namespaceId));
+    responder.sendString(HttpResponseStatus.OK, String.format("Deleted repository configuration for namespace '%s'.",
+                                                              namespaceId));
+  }
+
+  private NamespaceId validateNamespaceId(String namespaceId) throws BadRequestException {
+    try {
+      return new NamespaceId(namespaceId);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException("Namespace id can contain only alphanumeric characters or '_'.");
+    }
+  }
+
+  private RepositoryConfigRequest getRepository(FullHttpRequest request) throws BadRequestException {
+    try {
+      return parseBody(request, RepositoryConfigRequest.class);
+    } catch (JsonSyntaxException e) {
+      throw new BadRequestException("Invalid json object provided in request body.");
```

Comment:
Should not drop the root cause, otherwise there is no indication of what is wrong.

`throw new BadRequestException("Invalid request body: " + e.getMessage(), e);`